File: Java/roateArray.java
public class roateArray
{

static void revese(int[] arr, int start,int end)
{
	
	

	while( start < end)
	{
		int temp =arr[start];

		arr[start]=arr[end];

		arr[end]=temp;

		start ++;
		end--;
    }


  

}

 static void roate(int[] arr, int h)
   {

     
   	  revese(arr,0,h-1);
      revese(arr,h,arr.length-1);
      revese(arr,0,arr.length-1);

   }



	public static void main(String args[])
	{

         int[] arr = {1,2,3,4,5};

         int h=3;
           
         roate(arr,h);

        


         for(int x:arr)
         {
         	System.out.println(x);
         }
	}
}

File: Java/javaScript/roateArray.js
function name(va,start,end)
{
   while(start < end)
  {
    let temp=va[start]
    va[start]=va[end]
    va[end]=temp;
     start++
     end--
  }
  return va;
  
 }
 function roate(va)
 {
        var value=3;
        name(va,0,value-1);
        name(va,value,va.length-1)
        name(va,0,va.length-1)
        
        return va;
    
   
   
}
let va=roate([1,2,3,4,5,6,7,8,9])
console.log(va);